File: UVA/Math/10717.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef vector<ll> vi;
typedef pair<ll,ll> pi;
typedef vector<pi> vpi;
typedef double dl;
#define mp make_pair
#define pb push_back
#define f first
#define s second
#define MOD 1e9+7
#define lb lower_bound
#define ub upper_bound
#define dst distance
#define INF 1000000000

int gcd(int a, int b){
	if(a==0)return b;
	return gcd(b%a,a);
}

int lcmh(int a, int b){return a*b/gcd(a,b);}

int lcm(int a, int b, int c, int d){
	c = lcmh(min(c,d),max(c,d));
	b = lcmh(min(b,c),max(b,c));
	return lcmh(min(a,b),max(a,b));
}

int N,T,a;
vi V;

void run(int x){
	int low=0,high=INF;
	for(int i = 3; i < N; ++i){
		for(int j = 2; j < i; ++j){
			for(int k = 1; k < j; ++k){
				for(int l=0; l < k; ++l){
					int LCM = lcm(V[i],V[j],V[k],V[l]);
					if(x%LCM==0){
						cout << x << ' ' << x << '\n';
						return;
					}
					int t = x/LCM;
					low = max(low,t*LCM);
					high = min(high,(t+1)*LCM);
				}
			}
		}
	}
	cout << low << ' ' << high << '\n';
}

int main(){
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	//freopen("input.txt","r",stdin);
	while(cin>>N>>T){
		if(!N&&!T)return 0;
		V.clear();
		for(int i = 0; i < N; ++i){cin>>a;V.pb(a);}
		for(int i = 0; i < T; ++i){cin>>a;run(a);}
	}
}

File: UVA/Math/10717.py
def gcd(a,b): # a smaller
    if(a==0):
        return b
    return gcd(b%a,a)

def lcmh(a,b):
    return a*b/gcd(a,b)    

def lcm(a,b,c,d):
    c = lcmh(d,c)
    b = lcmh(min(b,c),max(b,c))
    a = lcmh(min(a,b),max(a,b))
    return a

from math import floor,inf

def run(A,V,i):
    bestlow = 0
    besthigh = inf
    for a in range(3,A[0]):
        for b in range(2,a):
            for c in range(1,b):
                for d in range(c):
                    l = lcm(V[a],V[b],V[c],V[d])
                    if(i%l == 0):
                        print(str(int(i)) + ' ' + str(int(i)))
                        return
                    else:
                        x = floor(i/l)
                        y = x+1
                    bestlow = max(bestlow,x*l)
                    besthigh = min(besthigh,y*l)
    print(str(int(bestlow)) + ' ' + str(int(besthigh)))

while(True):
    A = [int(i) for i in input().split(' ')]
    if (A == [0,0]):
        break
    V = []
    for i in range(A[0]):V.append(int(input()))
    V.sort()
    H = []
    for i in range(A[1]):H.append(int(input()))
    for x in H:
        run(A,V,x)